<html>
<head>
  <title>VEP output</title>
  <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
  <script src="http://ajax.aspnetcdn.com/ajax/jquery.dataTables/1.9.4/jquery.dataTables.min.js"></script>
  <script type="text/javascript" language="javascript">
    $(document).ready(function() {
      $('#data').dataTable({
        "sPaginationType": "full_numbers"
      });
    });
    
    function fnShowHide( iCol ) {
      /* Get the DataTables object again - this is not a recreation, just a get of the object */
      var oTable = $('#data').dataTable();
       
      var bVis = oTable.fnSettings().aoColumns[iCol].bVisible;
      oTable.fnSetColumnVis( iCol, bVis ? false : true );
    }
    
    function showAllCols() {
      var oTable = $('#data').dataTable();
      for (var i=0;i<oTable.fnSettings().aoColumns.length;i++) { 
        oTable.fnSetColumnVis(i, true);
      }
    }
    
    function showHide(lyr) {
      var lyrobj = document.getElementById(lyr);
      
      if(lyrobj.style.height == "0px") {
        lyrobj.style.height = "";
        lyrobj.style.display = "";
      }
      
      else {
        lyrobj.style.height = "0px";
        lyrobj.style.display = "none";
      }
    }
  </script>
  <style type="text/css">
    @import "http://www.datatables.net/release-datatables/media/css/demo_table.css";
    body {
      font-family: arial, sans-serif;
      margin: 0px;
      padding: 0px;
    }
    
    a {color: #36b;}
    a.visited {color: #006;}
    
    th {
      font-size: 11px;
    }
    td {
      font-size: 11px;
    }
    
    .masthead {
      background-color: rgb(51, 51, 102);
      color: rgb(204, 221, 255);
      height: 80px;
      width: 100%;
      padding: 0px;
    }
    
    .main {
      padding: 10px;
    }
    
    .gradient {
      background: #333366; /* Old browsers */
      background: -moz-linear-gradient(left,  #333366 0%, #ffffff 100%); /* FF3.6+ */
      background: -webkit-gradient(linear, left top, right top, color-stop(0%,#333366), color-stop(100%,#ffffff)); /* Chrome,Safari4+ */
      background: -webkit-linear-gradient(left,  #333366 0%,#ffffff 100%); /* Chrome10+,Safari5.1+ */
      background: -o-linear-gradient(left,  #333366 0%,#ffffff 100%); /* Opera 11.10+ */
      background: -ms-linear-gradient(left,  #333366 0%,#ffffff 100%); /* IE10+ */
      background: linear-gradient(to right,  #333366 0%,#ffffff 100%); /* W3C */
      filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#333366', endColorstr='#ffffff',GradientType=1 ); /* IE6-9 */
      
      padding: 0px;
      height: 80px;
      width: 500px;
      float: right;
      display: inline;
    }
  </style>
  </head>
  <body>
  <div id="masthead" class="masthead">
    <div style="float: left; display: inline; padding: 10px; height: 80px;">
      <a href="http://www.ensembl.org/"><img src="http://static.ensembl.org/i/e-ensembl.png"></a>
    </div>
    
    <div style="float: right; display: inline; height: 80px; background: white; padding: 10px;">
      <a href="http://www.ensembl.org/info/docs/variation/vep/vep_script.html"><img src="http://www.ensembl.org/img/vep_logo.png"></a>
    </div>
    <div class="gradient">
    </div>
  </div>
  <div class="main">
  <p>
    View: <a href="sample.vep.vcf_summary.html">Summary statistics</a> |
    <a href="sample.vep.vcf">as text</a> |
    <a href="javascript:void();" onclick="showHide('header')">Show/hide header</a> |
    <a href="javascript:void();" onclick="showAllCols()">Restore columns</a>
  </p>
  <hr/>
  <pre id="header" style="height:0px; display: none;">
##fileformat=VCFv4.1
##FILTER=<ID=PASS,Description="Accept as a confident somatic mutation">
##FILTER=<ID=REJECT,Description="Rejected as a confident somatic mutation">
##FORMAT=<ID=AD,Number=.,Type=Integer,Description="Allelic depths for the ref and alt alleles in the order listed">
##FORMAT=<ID=BQ,Number=A,Type=Float,Description="Average base quality for reads supporting alleles">
##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Approximate read depth (reads with MQ=255 or with bad mates are filtered)">
##FORMAT=<ID=FA,Number=A,Type=Float,Description="Allele fraction of the alternate allele with regard to reference">
##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
##FORMAT=<ID=PL,Number=G,Type=Integer,Description="Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification">
##FORMAT=<ID=SS,Number=1,Type=Integer,Description="Variant status relative to non-adjacent Normal,0=wildtype,1=germline,2=somatic,3=LOH,4=post-transcriptional modification,5=unknown">
##GATKCommandLine=<ID=MuTect,Version=3.1-0-g72492bb,Date="Tue Aug 30 21:38:46 UTC 2016",Epoch=1472593126856,CommandLineOptions="analysis_type=MuTect input_file=[/sbgenomics/Projects/0f55dae7-c301-4552-b7de-cdf27f76fe3c/JWCI-8-C_SRR479577_T.sort.edited.bam, /sbgenomics/Projects/0f55dae7-c301-4552-b7de-cdf27f76fe3c/JWCI-8-P_SRR479673_N.sort.edited.bam] showFullBamList=false read_buffer_size=null phone_home=AWS gatk_key=null tag=NA read_filter=[] intervals=[/sbgenomics/Projects/0f55dae7-c301-4552-b7de-cdf27f76fe3c/exome_targets_chrM_Y_removed.b37.bed] excludeIntervals=null interval_set_rule=UNION interval_merging=ALL interval_padding=0 reference_sequence=/sbgenomics/Projects/0f55dae7-c301-4552-b7de-cdf27f76fe3c/Homo_sapiens_assembly19_chrY_M_removed.fasta nonDeterministicRandomSeed=false disableDithering=false maxRuntime=-1 maxRuntimeUnits=MINUTES downsampling_type=BY_SAMPLE downsample_to_fraction=null downsample_to_coverage=1000 baq=OFF baqGapOpenPenalty=40.0 fix_misencoded_quality_scores=false allow_potentially_misencoded_quality_scores=false useOriginalQualities=false defaultBaseQualities=-1 performanceLog=null BQSR=null quantize_quals=0 disable_indel_quals=false emit_original_quals=false preserve_qscores_less_than=6 globalQScorePrior=-1.0 validation_strictness=SILENT remove_program_records=false keep_program_records=false sample_rename_mapping_file=null unsafe=null disable_auto_index_creation_and_locking_when_reading_rods=false num_threads=1 num_cpu_threads_per_data_thread=1 num_io_threads=0 monitorThreadEfficiency=false num_bam_file_handles=null read_group_black_list=null pedigree=[] pedigreeString=[] pedigreeValidationType=STRICT allow_intervals_with_unindexed_bam=false generateShadowBCF=false variant_index_type=DYNAMIC_SEEK variant_index_parameter=-1 logging_level=INFO log_to_file=null help=false version=false noop=false enable_extended_output=false enable_qscore_output=false artifact_detection_mode=false tumor_sample_name=TUMOR bam_tumor_sample_name=null normal_sample_name=NORMAL force_output=false force_alleles=false only_passing_calls=false initial_tumor_lod=4.0 tumor_lod=6.3 fraction_contamination=0.02 minimum_mutation_cell_fraction=0.0 normal_lod=2.2 normal_artifact_lod=1.0 strand_artifact_lod=2.0 strand_artifact_power_threshold=0.9 dbsnp_normal_lod=5.5 minimum_normal_allele_fraction=0.0 tumor_f_pretest=0.005 min_qscore=5 gap_events_threshold=3 heavily_clipped_read_fraction=0.3 fraction_mapq0_threshold=0.5 pir_median_threshold=10.0 pir_mad_threshold=3.0 required_maximum_alt_allele_mapping_quality_score=20 max_alt_alleles_in_normal_count=2 max_alt_alleles_in_normal_qscore_sum=20 max_alt_allele_in_normal_fraction=0.03 power_constant_qscore=30 power_constant_af=0.30000001192092896 vcf=org.broadinstitute.sting.gatk.io.stubs.VariantContextWriterStub no_cmdline_in_header=org.broadinstitute.sting.gatk.io.stubs.VariantContextWriterStub sites_only=org.broadinstitute.sting.gatk.io.stubs.VariantContextWriterStub bcf=org.broadinstitute.sting.gatk.io.stubs.VariantContextWriterStub dbsnp=[] cosmic=[] normal_panel=[] coverage_file=null coverage_20_q20_file=null power_file=null tumor_depth_file=null normal_depth_file=null filter_reads_with_N_cigar=false filter_mismatching_base_and_quals=false filter_bases_not_stored=false">
##INFO=<ID=DB,Number=0,Type=Flag,Description="dbSNP Membership">
##INFO=<ID=MQ0,Number=1,Type=Integer,Description="Total Mapping Quality Zero Reads">
##INFO=<ID=SOMATIC,Number=0,Type=Flag,Description="Somatic event">
##INFO=<ID=VT,Number=1,Type=String,Description="Variant type, can be SNP, INS or DEL">
##contig=<ID=1,length=249250621>
##contig=<ID=2,length=243199373>
##contig=<ID=3,length=198022430>
##contig=<ID=4,length=191154276>
##contig=<ID=5,length=180915260>
##contig=<ID=6,length=171115067>
##contig=<ID=7,length=159138663>
##contig=<ID=8,length=146364022>
##contig=<ID=9,length=141213431>
##contig=<ID=10,length=135534747>
##contig=<ID=11,length=135006516>
##contig=<ID=12,length=133851895>
##contig=<ID=13,length=115169878>
##contig=<ID=14,length=107349540>
##contig=<ID=15,length=102531392>
##contig=<ID=16,length=90354753>
##contig=<ID=17,length=81195210>
##contig=<ID=18,length=78077248>
##contig=<ID=19,length=59128983>
##contig=<ID=20,length=63025520>
##contig=<ID=21,length=48129895>
##contig=<ID=22,length=51304566>
##contig=<ID=X,length=155270560>
##contig=<ID=GL000207.1,length=4262>
##contig=<ID=GL000226.1,length=15008>
##contig=<ID=GL000229.1,length=19913>
##contig=<ID=GL000231.1,length=27386>
##contig=<ID=GL000210.1,length=27682>
##contig=<ID=GL000239.1,length=33824>
##contig=<ID=GL000235.1,length=34474>
##contig=<ID=GL000201.1,length=36148>
##contig=<ID=GL000247.1,length=36422>
##contig=<ID=GL000245.1,length=36651>
##contig=<ID=GL000197.1,length=37175>
##contig=<ID=GL000203.1,length=37498>
##contig=<ID=GL000246.1,length=38154>
##contig=<ID=GL000249.1,length=38502>
##contig=<ID=GL000196.1,length=38914>
##contig=<ID=GL000248.1,length=39786>
##contig=<ID=GL000244.1,length=39929>
##contig=<ID=GL000238.1,length=39939>
##contig=<ID=GL000202.1,length=40103>
##contig=<ID=GL000234.1,length=40531>
##contig=<ID=GL000232.1,length=40652>
##contig=<ID=GL000206.1,length=41001>
##contig=<ID=GL000240.1,length=41933>
##contig=<ID=GL000236.1,length=41934>
##contig=<ID=GL000241.1,length=42152>
##contig=<ID=GL000243.1,length=43341>
##contig=<ID=GL000242.1,length=43523>
##contig=<ID=GL000230.1,length=43691>
##contig=<ID=GL000237.1,length=45867>
##contig=<ID=GL000233.1,length=45941>
##contig=<ID=GL000204.1,length=81310>
##contig=<ID=GL000198.1,length=90085>
##contig=<ID=GL000208.1,length=92689>
##contig=<ID=GL000191.1,length=106433>
##contig=<ID=GL000227.1,length=128374>
##contig=<ID=GL000228.1,length=129120>
##contig=<ID=GL000214.1,length=137718>
##contig=<ID=GL000221.1,length=155397>
##contig=<ID=GL000209.1,length=159169>
##contig=<ID=GL000218.1,length=161147>
##contig=<ID=GL000220.1,length=161802>
##contig=<ID=GL000213.1,length=164239>
##contig=<ID=GL000211.1,length=166566>
##contig=<ID=GL000199.1,length=169874>
##contig=<ID=GL000217.1,length=172149>
##contig=<ID=GL000216.1,length=172294>
##contig=<ID=GL000215.1,length=172545>
##contig=<ID=GL000205.1,length=174588>
##contig=<ID=GL000219.1,length=179198>
##contig=<ID=GL000224.1,length=179693>
##contig=<ID=GL000223.1,length=180455>
##contig=<ID=GL000195.1,length=182896>
##contig=<ID=GL000212.1,length=186858>
##contig=<ID=GL000222.1,length=186861>
##contig=<ID=GL000200.1,length=187035>
##contig=<ID=GL000193.1,length=189789>
##contig=<ID=GL000194.1,length=191469>
##contig=<ID=GL000225.1,length=211173>
##contig=<ID=GL000192.1,length=547496>
##contig=<ID=NC_007605,length=171823>
##reference=file:///sbgenomics/Projects/0f55dae7-c301-4552-b7de-cdf27f76fe3c/Homo_sapiens_assembly19_chrY_M_removed.fasta
##VEP=v83 cache=/opt/variant_effect_predictor_83/cache/homo_sapiens/83_GRCh37 db=. polyphen=2.2.2 HGMD-PUBLIC=20152 sift=sift5.2.2 ESP=20141103 genebuild=2011-04 ClinVar=201507 COSMIC=71 gencode=GENCODE 19 regbuild=13 assembly=GRCh37.p13 dbSNP=144
##INFO=<ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|SYMBOL_SOURCE|HGNC_ID|FREQS|CLIN_SIG|SOMATIC|PHENO">
</pre><table id="data" class="display"><thead><tr><th>#CHROM <a onclick="fnShowHide(0);" href="javascript:void();"><img width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" height="6" /></a></th><th>POS <a href="javascript:void();" onclick="fnShowHide(1);"><img height="6" width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" /></a></th><th>ID <a onclick="fnShowHide(2);" href="javascript:void();"><img height="6" width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" /></a></th><th>REF <a href="javascript:void();" onclick="fnShowHide(3);"><img width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" height="6" /></a></th><th>ALT <a href="javascript:void();" onclick="fnShowHide(4);"><img style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" width="6" height="6" /></a></th><th>QUAL <a onclick="fnShowHide(5);" href="javascript:void();"><img height="6" src="http://www.ensembl.org/i/16/cross.png" style="border: 1px solid gray; padding: 1px;" width="6" /></a></th><th>FILTER <a href="javascript:void();" onclick="fnShowHide(6);"><img width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" height="6" /></a></th><th>INFO <a href="javascript:void();" onclick="fnShowHide(7);"><img width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" height="6" /></a></th><th>FORMAT <a onclick="fnShowHide(8);" href="javascript:void();"><img height="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" width="6" /></a></th><th>TUMOR <a onclick="fnShowHide(9);" href="javascript:void();"><img height="6" src="http://www.ensembl.org/i/16/cross.png" style="border: 1px solid gray; padding: 1px;" width="6" /></a></th><th>NORMAL <a onclick="fnShowHide(10);" href="javascript:void();"><img src="http://www.ensembl.org/i/16/cross.png" style="border: 1px solid gray; padding: 1px;" width="6" height="6" /></a></th></thead></tr><tbody></tbody><tfoot><tr><th>#CHROM <a onclick="fnShowHide(0);" href="javascript:void();"><img width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" height="6" /></a></th><th>POS <a href="javascript:void();" onclick="fnShowHide(1);"><img height="6" width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" /></a></th><th>ID <a onclick="fnShowHide(2);" href="javascript:void();"><img height="6" width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" /></a></th><th>REF <a href="javascript:void();" onclick="fnShowHide(3);"><img width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" height="6" /></a></th><th>ALT <a href="javascript:void();" onclick="fnShowHide(4);"><img style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" width="6" height="6" /></a></th><th>QUAL <a onclick="fnShowHide(5);" href="javascript:void();"><img height="6" src="http://www.ensembl.org/i/16/cross.png" style="border: 1px solid gray; padding: 1px;" width="6" /></a></th><th>FILTER <a href="javascript:void();" onclick="fnShowHide(6);"><img width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" height="6" /></a></th><th>INFO <a href="javascript:void();" onclick="fnShowHide(7);"><img width="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" height="6" /></a></th><th>FORMAT <a onclick="fnShowHide(8);" href="javascript:void();"><img height="6" style="border: 1px solid gray; padding: 1px;" src="http://www.ensembl.org/i/16/cross.png" width="6" /></a></th><th>TUMOR <a onclick="fnShowHide(9);" href="javascript:void();"><img height="6" src="http://www.ensembl.org/i/16/cross.png" style="border: 1px solid gray; padding: 1px;" width="6" /></a></th><th>NORMAL <a onclick="fnShowHide(10);" href="javascript:void();"><img src="http://www.ensembl.org/i/16/cross.png" style="border: 1px solid gray; padding: 1px;" width="6" height="6" /></a></th></tr></tfoot></table><p>&nbsp;</p></div></html></body>
</html>
