<html>
<head>
  <title>VEP summary</title>
  <script type="text/javascript" src="http://www.google.com/jsapi"></script>
  <script type="text/javascript">
    google.load('visualization', '1', {packages: ['corechart','table']});
  </script>
  <script type="text/javascript">
    
    function init() {
      // charts
      var var_class_pie = drawPie('var_class_pie', 'Variant classes', google.visualization.arrayToDataTable([['Variant class','Count'],['SNV',5351]]), null);
var var_class_table = drawTable('var_class_table', 'Variant classes', google.visualization.arrayToDataTable([['Variant class','Count'],['SNV',5351]]));

            google.visualization.events.addListener(var_class_pie, 'select', function() {
              var_class_table.setSelection(var_class_pie.getSelection());
            });
            google.visualization.events.addListener(var_class_table, 'select', function() {
              var_class_pie.setSelection(var_class_table.getSelection());
            });
          var var_cons_pie = drawPie('var_cons_pie', 'Consequences (most severe)', google.visualization.arrayToDataTable([['Consequence type','Count'],['splice_donor_variant',30],['splice_acceptor_variant',19],['stop_gained',104],['stop_lost',5],['missense_variant',2160],['splice_region_variant',147],['stop_retained_variant',3],['synonymous_variant',689],['5_prime_UTR_variant',91],['3_prime_UTR_variant',240],['non_coding_transcript_exon_variant',554],['intron_variant',1251],['upstream_gene_variant',35],['downstream_gene_variant',17],['intergenic_variant',6]]), {slices: [{color: "#ff7f50"}, {color: "#ff7f50"}, {color: "#ff0000"}, {color: "#ff0000"}, {color: "#ffd700"}, {color: "#ff7f50"}, {color: "#76ee00"}, {color: "#76ee00"}, {}, {}, {}, {color: "#02599c"}, {color: "#a2b5cd"}, {color: "#a2b5cd"}, {color: "gray"}]});
var var_cons_table = drawTable('var_cons_table', 'Consequences (most severe)', google.visualization.arrayToDataTable([['Consequence type','Count'],['splice_donor_variant',30],['splice_acceptor_variant',19],['stop_gained',104],['stop_lost',5],['missense_variant',2160],['splice_region_variant',147],['stop_retained_variant',3],['synonymous_variant',689],['5_prime_UTR_variant',91],['3_prime_UTR_variant',240],['non_coding_transcript_exon_variant',554],['intron_variant',1251],['upstream_gene_variant',35],['downstream_gene_variant',17],['intergenic_variant',6]]));

            google.visualization.events.addListener(var_cons_pie, 'select', function() {
              var_cons_table.setSelection(var_cons_pie.getSelection());
            });
            google.visualization.events.addListener(var_cons_table, 'select', function() {
              var_cons_pie.setSelection(var_cons_table.getSelection());
            });
          var consequences_pie = drawPie('consequences_pie', 'Consequences (all)', google.visualization.arrayToDataTable([['Consequence type','Count'],['splice_acceptor_variant',84],['splice_donor_variant',150],['stop_gained',318],['stop_lost',7],['missense_variant',6159],['splice_region_variant',847],['incomplete_terminal_codon_variant',1],['stop_retained_variant',6],['synonymous_variant',1975],['coding_sequence_variant',2],['5_prime_UTR_variant',355],['3_prime_UTR_variant',1168],['non_coding_transcript_exon_variant',3336],['intron_variant',9965],['NMD_transcript_variant',1598],['non_coding_transcript_variant',6194],['upstream_gene_variant',4333],['downstream_gene_variant',6449],['intergenic_variant',6]]), {slices: [{color: "#ff7f50"}, {color: "#ff7f50"}, {color: "#ff0000"}, {color: "#ff0000"}, {color: "#ffd700"}, {color: "#ff7f50"}, {color: "#ff00ff"}, {color: "#76ee00"}, {color: "#76ee00"}, {color: "#458b00"}, {}, {}, {}, {color: "#02599c"}, {}, {}, {color: "#a2b5cd"}, {color: "#a2b5cd"}, {color: "gray"}]});
var consequences_table = drawTable('consequences_table', 'Consequences (all)', google.visualization.arrayToDataTable([['Consequence type','Count'],['splice_acceptor_variant',84],['splice_donor_variant',150],['stop_gained',318],['stop_lost',7],['missense_variant',6159],['splice_region_variant',847],['incomplete_terminal_codon_variant',1],['stop_retained_variant',6],['synonymous_variant',1975],['coding_sequence_variant',2],['5_prime_UTR_variant',355],['3_prime_UTR_variant',1168],['non_coding_transcript_exon_variant',3336],['intron_variant',9965],['NMD_transcript_variant',1598],['non_coding_transcript_variant',6194],['upstream_gene_variant',4333],['downstream_gene_variant',6449],['intergenic_variant',6]]));

            google.visualization.events.addListener(consequences_pie, 'select', function() {
              consequences_table.setSelection(consequences_pie.getSelection());
            });
            google.visualization.events.addListener(consequences_table, 'select', function() {
              consequences_pie.setSelection(consequences_table.getSelection());
            });
          var coding_pie = drawPie('coding_pie', 'Coding consequences', google.visualization.arrayToDataTable([['Consequence type','Count'],['stop_gained',318],['stop_lost',7],['missense_variant',6159],['synonymous_variant',1975],['coding_sequence_variant',2]]), {slices: [{color: "#ff0000"}, {color: "#ff0000"}, {color: "#ffd700"}, {color: "#76ee00"}, {color: "#458b00"}]});
var coding_table = drawTable('coding_table', 'Coding consequences', google.visualization.arrayToDataTable([['Consequence type','Count'],['stop_gained',318],['stop_lost',7],['missense_variant',6159],['synonymous_variant',1975],['coding_sequence_variant',2]]));

            google.visualization.events.addListener(coding_pie, 'select', function() {
              coding_table.setSelection(coding_pie.getSelection());
            });
            google.visualization.events.addListener(coding_table, 'select', function() {
              coding_pie.setSelection(coding_table.getSelection());
            });
          var chr_bar = drawBar('chr_bar', 'Variants by chromosome', google.visualization.arrayToDataTable([['Chromosome','Count'],['1',5351]]), {legend: {position: "none"}});
var chr_table = drawTable('chr_table', 'Variants by chromosome', google.visualization.arrayToDataTable([['Chromosome','Count'],['1',5351]]));

            google.visualization.events.addListener(chr_bar, 'select', function() {
              chr_table.setSelection(chr_bar.getSelection());
            });
            google.visualization.events.addListener(chr_table, 'select', function() {
              chr_bar.setSelection(chr_table.getSelection());
            });
          var chr_1_area = drawArea('chr_1_area', 'Distribution of variants on chromosome 1', google.visualization.arrayToDataTable([['Position (mb)','Count'],['0',25],['1',92],['2',25],['3',24],['4',0],['5',2],['6',33],['7',14],['8',11],['9',30],['10',22],['11',43],['12',160],['13',107],['14',7],['15',22],['16',294],['17',115],['18',10],['19',45],['20',19],['21',45],['22',63],['23',29],['24',33],['25',13],['26',44],['27',38],['28',24],['29',17],['30',0],['31',15],['32',42],['33',38],['34',13],['35',25],['36',48],['37',3],['38',43],['39',37],['40',36],['41',11],['42',14],['43',37],['44',34],['45',47],['46',30],['47',21],['48',7],['49',0],['50',2],['51',5],['52',21],['53',22],['54',20],['55',29],['56',0],['57',11],['58',1],['59',2],['60',10],['61',2],['62',13],['63',3],['64',9],['65',18],['66',3],['67',17],['68',10],['69',0],['70',11],['71',0],['72',3],['73',0],['74',6],['75',10],['76',8],['77',1],['78',20],['79',5],['80',0],['81',0],['82',2],['83',0],['84',6],['85',16],['86',12],['87',6],['88',0],['89',13],['90',5],['91',22],['92',24],['93',11],['94',18],['95',9],['96',0],['97',4],['98',2],['99',3],['100',12],['101',4],['102',1],['103',7],['104',3],['105',0],['106',0],['107',4],['108',9],['109',42],['110',51],['111',28],['112',17],['113',11],['114',20],['115',11],['116',15],['117',21],['118',16],['119',4],['120',39],['121',8],['122',0],['123',0],['124',0],['125',0],['126',0],['127',0],['128',0],['129',0],['130',0],['131',0],['132',0],['133',0],['134',0],['135',0],['136',0],['137',0],['138',0],['139',0],['140',0],['141',0],['142',13],['143',18],['144',176],['145',202],['146',74],['147',11],['148',142],['149',30],['150',57],['151',65],['152',144],['153',70],['154',59],['155',69],['156',96],['157',19],['158',49],['159',26],['160',60],['161',50],['162',21],['163',1],['164',5],['165',14],['166',5],['167',16],['168',11],['169',24],['170',7],['171',11],['172',4],['173',14],['174',4],['175',9],['176',13],['177',10],['178',7],['179',26],['180',18],['181',7],['182',14],['183',34],['184',8],['185',9],['186',35],['187',0],['188',0],['189',0],['190',1],['191',0],['192',3],['193',4],['194',0],['195',0],['196',18],['197',19],['198',7],['199',0],['200',23],['201',37],['202',26],['203',30],['204',26],['205',38],['206',29],['207',29],['208',14],['209',22],['210',11],['211',13],['212',15],['213',8],['214',31],['215',11],['216',12],['217',2],['218',3],['219',2],['220',22],['221',2],['222',22],['223',13],['224',13],['225',8],['226',26],['227',27],['228',35],['229',9],['230',14],['231',18],['232',5],['233',8],['234',8],['235',29],['236',37],['237',11],['238',0],['239',0],['240',9],['241',11],['242',6],['243',10],['244',8],['245',8],['246',9],['247',32],['248',109],['249',6]]), {hAxis: {title: "Position (mb)", textStyle: {fontSize: 8}}, legend: {position: "none"}});
var protein_bar = drawBar('protein_bar', 'Position in protein', google.visualization.arrayToDataTable([['Position in protein (percentile)','Count'],['00-10%',837],['10-20%',774],['20-30%',812],['30-40%',822],['40-50%',894],['50-60%',824],['60-70%',793],['70-80%',879],['80-90%',865],['90-100%',967]]), {hAxis: {title: "Position in protein (percentile)", textStyle: {fontSize: 10}}, legend: {position: "none"}});

    }
    
    function drawPie(id, title, data, options) {    
      var pie = new google.visualization.PieChart(document.getElementById(id));
      pie.draw(data, options);
      return pie;
    }
    function drawBar(id, title, data, options) {
      var bar = new google.visualization.ColumnChart(document.getElementById(id));
      bar.draw(data, options);
      return bar;
    }
    function drawTable(id, title, data) {
      var table = new google.visualization.Table(document.getElementById(id));
      table.draw(data, null);
      return table;
    }
    function drawLine(id, title, data, options) {
      var line = new google.visualization.LineChart(document.getElementById(id));
      line.draw(data, options);
      return line;
    }
    function drawArea(id, title, data, options) {
      var area = new google.visualization.AreaChart(document.getElementById(id));
      area.draw(data, options);
      return area;
    }
    google.setOnLoadCallback(init);
  </script>
  
  
  <style type="text/css">
    body {
      font-family: arial, sans-serif;
      margin: 0px;
      padding: 0px;
    }
    
    a {color: #36b;}
    a.visited {color: #006;}
    
    .stats_table {
      margin: 5px;
    }
    
    tr:nth-child(odd) {
      background-color: rgb(238, 238, 238);
    }
    
    td {
      padding: 5px;
    }
    
    td:nth-child(odd) {
      font-weight: bold;
    }
    
    h3 {
      color: #666;
    }
    
    .masthead {
      background-color: rgb(51, 51, 102);
      color: rgb(204, 221, 255);
      height: 80px;
      width: 100%;
      padding: 0px;
    }
    
    .main {
      padding: 10px;
    }
    
    .gradient {
      background: #333366; /* Old browsers */
      background: -moz-linear-gradient(left,  #333366 0%, #ffffff 100%); /* FF3.6+ */
      background: -webkit-gradient(linear, left top, right top, color-stop(0%,#333366), color-stop(100%,#ffffff)); /* Chrome,Safari4+ */
      background: -webkit-linear-gradient(left,  #333366 0%,#ffffff 100%); /* Chrome10+,Safari5.1+ */
      background: -o-linear-gradient(left,  #333366 0%,#ffffff 100%); /* Opera 11.10+ */
      background: -ms-linear-gradient(left,  #333366 0%,#ffffff 100%); /* IE10+ */
      background: linear-gradient(to right,  #333366 0%,#ffffff 100%); /* W3C */
      filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#333366', endColorstr='#ffffff',GradientType=1 ); /* IE6-9 */
      
      padding: 0px;
      height: 80px;
      width: 500px;
      float: right;
      display: inline;
    }
    
    .main_content {
      margin-left: 300px;
    }
    
    .sidemenu {
      width: 260px;
      position: fixed;
      border-style: solid;
      border-width: 2px;
      border-color: rgb(51, 51, 102);
    }
    
    .sidemenu_head {
      width: 250px;
      background-color: rgb(51, 51, 102);
      color: rgb(204, 221, 255);
      padding: 5px;
    }
    
    .sidemenu_body {
      width: 250px;
      padding: 5px;
    }
  </style>
</head>
<body>
<div id="masthead" class="masthead">
  <div style="float: left; display: inline; padding: 10px; height: 80px;">
    <a href="http://www.ensembl.org/"><img src="http://static.ensembl.org/i/e-ensembl.png"></a>
  </div>
  
  <div style="float: right; display: inline; height: 80px; background: white; padding: 10px;">
    <a href="http://www.ensembl.org/info/docs/variation/vep/vep_script.html"><img src="http://www.ensembl.org/img/vep_logo.png"></a>
  </div>
  <div class="gradient">
  </div>
</div>
<div class="main">
<div class="sidemenu"><div class="sidemenu_head">Links</div> <div class="sidemenu_body"><ul><li><a href="#masthead">Top of page</a></li> <li><a href="#run_stats">VEP run statistics</a></li> <li><a href="#gen_stats">General statistics</a></li> <li><a href="#var_class">Variant classes</a></li> <li><a href="#var_cons">Consequences (most severe)</a></li> <li><a href="#consequences">Consequences (all)</a></li> <li><a href="#coding">Coding consequences</a></li> <li><a href="#chr">Variants by chromosome</a></li> <li><a href="#protein">Position in protein</a></li></ul></div></div><div class='main_content'><h3 id="run_stats">VEP run statistics</h3><table class="stats_table"><tr><td>VEP version (API)</td> <td>83 (83)</td></tr> <tr><td>Cache/Database</td> <td>/opt/variant_effect_predictor_83/cache/homo_sapiens/83_GRCh37</td></tr> <tr><td>Species</td> <td>homo_sapiens</td></tr> <tr><td>Command line options</td> <td><pre>--input_file /mnt/data/cloudneo/sample.vcf --dir /opt/variant_effect_predictor_83/cache --offline --output_file sample.vep.vcf --stats_file sample.vep.vcf_summary.html --filter_common --html --vcf</pre></td></tr> <tr><td>Start time</td> <td>2017-03-07 18:04:27</td></tr> <tr><td>End time</td> <td>2017-03-07 18:15:11</td></tr> <tr><td>Run time</td> <td>644 seconds</td></tr> <tr><td>Input file (format)</td> <td>/mnt/data/cloudneo/sample.vcf (VCF)</td></tr> <tr><td>Output file</td> <td>sample.vep.vcf <a href="sample.vep.vcf.html">[HTML]</a> <a href="sample.vep.vcf">[text]</a></td></tr></table><h3 id="gen_stats">General statistics</h3><table class="stats_table"><tr><td>Lines of input read</td> <td>88649</td></tr> <tr><td>Variants processed</td> <td>10424</td></tr> <tr><td>Variants remaining after filtering</td> <td>5351</td></tr> <tr><td>Lines of output written</td> <td>5351</td></tr> <tr><td>Novel / existing variants</td> <td>6719 (64.5%) / 3705 (35.5%)</td></tr> <tr><td>Overlapped genes</td> <td>2189</td></tr> <tr><td>Overlapped transcripts</td> <td>10468</td></tr> <tr><td>Overlapped regulatory features</td> <td>-</td></tr></table><hr /><h3 id="var_class">Variant classes</h3><div style="width: 800px; height: 200px" id="var_class_pie">&nbsp;</div><div id="var_class_table" style="width: 800px; height: 200px">&nbsp;</div><hr /><h3 id="var_cons">Consequences (most severe)</h3><div style="width: 800px; height: 400px" id="var_cons_pie">&nbsp;</div><div id="var_cons_table" style="width: 800px; height: 200px">&nbsp;</div><hr /><h3 id="consequences">Consequences (all)</h3><div id="consequences_pie" style="width: 800px; height: 400px">&nbsp;</div><div style="width: 800px; height: 200px" id="consequences_table">&nbsp;</div><hr /><h3 id="coding">Coding consequences</h3><div id="coding_pie" style="width: 800px; height: 400px">&nbsp;</div><div style="width: 800px; height: 200px" id="coding_table">&nbsp;</div><hr /><h3 id="chr">Variants by chromosome</h3><div style="width: 800px; height: 200px" id="chr_bar">&nbsp;</div><div id="chr_table" style="width: 800px; height: 200px">&nbsp;</div><hr /><h3 id="chr_1">Distribution of variants on chromosome 1</h3><div id="chr_1_area" style="width: 800px; height: 200px">&nbsp;</div><hr /><h3 id="protein">Position in protein</h3><div id="protein_bar" style="width: 800px; height: 200px">&nbsp;</div></div>
</div></body>
</html>
